src/v/cloud_storage/segment_chunk_api.h

```diff
@@ -73,6 +73,8 @@ class segment_chunks {
     iterator_t begin();
     iterator_t end();
 
+    const chunk_map_t& chunk_map() const { return _chunks; }
```

Comment:
Is there a descriptor anywhere about what this map represents (or more broadly, what segment_chunks is)? Is this a sparse map of only pending chunks? Or the entire segment sliced up into ranges? Does that map only grow larger, as chunks are downloaded?

Would appreciate a comment here or for the class.

Reply:
Added a comment. Normally this is an internal detail of the chunk API and should not be exposed (although it is safe to expose). An alternative would have been a method on the chunk API which calculates the segment range, given a prefetch value.

This change is required in this PR because the remote segment needs to do prefetch calculation which requires the map content. 

It will go away in the next PR because we can fix the prefetch calculation to follow the plan described in the cover letter. With those future changes the segment range will not be required and we will be able to just add prefetch related downloads to the chunk waiter list.

Reply:
This map is static and is created when the chunk API starts, and it does not change afterwards. It contains a mapping from chunk start offset to metadata, which is quite similar to the partition manifest mapping.

Reply:
Got it, thanks for the detail here. Yeah I figured this was likely a temporary solution. I'm fine with leaving this as is if so.

Reply:
this is now removed in the follow up PR https://github.com/redpanda-data/redpanda/pull/18278/commits/d6a6c13fa7c2902ce4306a87c676c3414a874585